src/command_modules/azure-cli-datalake/azure/cli/command_modules/datalake/store/custom.py

```diff
@@ -0,0 +1,257 @@
+
+    if tier:
+        create_params.new_tier = tier
+    if not disable_encryption:
+        identity = EncryptionIdentity()
+        config = EncryptionConfig()
+        if not encryption_type:
+            logger.warning('No Encryption type passed in, defaulting to Service managed encryption. To opt out, explicitly pass in --disable_encryption')
+            encryption_type = EncryptionConfigType.service_managed
+        if encryption_type == EncryptionConfigType.user_managed:
+            if not key_name or not key_vault_id or not key_version:
+                raise CLIError('For user managed encryption, --key_vault_id, --key_name and --key_version are required parameters and must be supplied.')
+            config.key_vault_meta_info = KeyVaultMetaInfo(key_vault_id, key_name, key_version)
+        else:
+            if key_name or key_vault_id or key_version:
+                logger.warning('User supplied Key Vault information. For service managed encryption user supplied Key Vault information is ignored.')
+        
+        config.type = encryption_type
+        create_params.encryption_config = config
+        create_params.identity = identity
+    else:
+        create_params.encryption_state = EncryptionState.disabled
+        create_params.identity = None
+        create_params.encryption_config = None
+
+    return client.create(resource_group_name, account_name, create_params)
+
+def update_adls_account(client,
+                        account_name,
+                        tags = None,
+                        resource_group_name = None,
+                        default_group = None,
+                        firewall_state = None,
+                        allow_azure_ips = None,
+                        trusted_id_provider_state = None,
+                        tier = None):
+    if not resource_group_name:
+        resource_group_name = _get_resource_group_by_account_name(client, account_name)
+
+    update_params = DataLakeStoreAccountUpdateParameters()
+
+    if tags:
+        update_params.tags = tags
+
+    if default_group:
+        update_params.default_group = default_group
+
+    if firewall_state:
+        update_params.firewall_state = firewall_state
+
+    if allow_azure_ips:
+        update_params.allow_azure_ips = allow_azure_ips
+
+    if trusted_id_provider_state:
+        update_params.trusted_id_provider_state = trusted_id_provider_state
+
+    if tier:
+        update_params.new_tier = tier
+
+    return client.update(resource_group_name, account_name, update_params)
+
+# firewall customizations
+def add_adls_firewall_rule(client,
+                           account_name,
+                           firewall_rule_name,
+                           start_ip_address,
+                           end_ip_address,
+                           resource_group_name = None):
+    if not resource_group_name:
+        resource_group_name = _get_resource_group_by_account_name(client, account_name)
```

Comment:
You could abstract this out to a validator and attach it to resource_group_name. Then this would happen automatically for this parameter at whatever scope you define.  (However, this is also just fine).